addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/BackupFileDisclosureScanRule.java

```diff
@@ -459,9 +459,12 @@ private void findBackupFile(HttpMessage originalMessage) throws Exception {
                 // a parent folder to mess with)
                 String[] temppathbreak = pathbreak;
                 String parentfoldername = pathbreak[pathbreak.length - 2];
+                // If the parent folder name is really short a collision is likely
+                // Default to a reasonable length, which may have the inverse effect but we'll
+                // chance it
+                int len = parentfoldername.length() >= 4 ? parentfoldername.length() : 4;
```

Comment:
Is reading this any easier or more clear in the opposite condition?
```suggestion
                int len = parentfoldername.length() <= 4 ? 4: parentfoldername.length();
```

Reply:
Could use `Math.max(parentfoldername.length(), 4);`